benchmarks/bench_pickle.py

```diff
@@ -0,0 +1,458 @@
+        . Compression levels   : {2}
+        . Compression algorithm: {3}
+        . Memory map mode      : {4}
+        . Bench nifti data     : {5}
+        . Bench big array      : {6}
+        . Bench 2 big arrays   : {7}
+        . Bench big dictionnary: {8}
+        . Bench array+dict     : {9}
+""".format(joblib.__version__,
+           args.tries,
+           ", ".join(map(str, args.compress)),
+           "Raw" if not args.compress else args.cname,
+           args.mmap,
+           args.nifti,
+           args.array,
+           args.arrays,
+           args.dict,
+           args.combo)
+
+    if args.array:
+        shape = tuple(args.shape)
+        size = round(np.multiply.reduce(shape) * 8 / 1024 ** 2, 1)
+        summary += """
+    - Big array:
+        . shape: {0}
+        . size in memory: {1} MB
+""".format(str(shape), size)
+
+    if args.dict:
+        summary += """
+    - Big dictionnary:
+        . number of keys: {0}
+        . value type: {1}
+""".format(args.size, 'np.ndarray'
+           if args.valuearray else 'str'
+            if args.valuestring else 'int')
+        if args.valuearray:
+            summary += """        . arrays shape: {0}
+""".format(str(tuple(args.valuearrayshape)))
+
+    if args.list:
+        summary += """
+    - Big list:
+        . number of elements: {0}
+        . value type: {1}
+""".format(args.size, 'np.ndarray'
+           if args.valuearray else 'str'
+            if args.valuestring else 'int')
+        if args.valuearray:
+            summary += """        . arrays shape: {0}
+""".format(str(tuple(args.valuearrayshape)))
+
+    print(summary)
+
+
+def bench_compress(dataset, name='',
+                   compress=None, cname='joblib', cache_size=0, tries=5):
+    """Bench joblib dump and load functions, compress modes."""
+    time_write = list()
+    time_read = list()
+    du = list()
+    mem_read = list()
+    mem_write = list()
+    clear_out()
+    time_write, obj = timeit(joblib.dump, dataset, 'out/test.pkl',
+                             tries=tries,
+                             compress=compress, cache_size=cache_size,
+                             cname=cname)
+    del obj
+    gc.collect()
+    mem_write = memory_used(joblib.dump, dataset, 'out/test.pkl',
+                            compress=compress, cache_size=cache_size,
+                            cname=cname)
+    delete_obj(dataset)
+    du = disk_used('out') / 1024.
+    time_read, obj = timeit(joblib.load, 'out/test.pkl', tries=tries)
+    delete_obj(obj)
+    mem_read = memory_used(joblib.load, 'out/test.pkl')
+    compress_str = "Raw" if compress is None else "%s %s" % (cname, compress)
+    print_line(name, compress_str, time_write, time_read,
+               mem_write, mem_read, du)
+
+
+def bench_mmap(dataset, name='', cache_size=0, mmap_mode='r', tries=5):
+    """Bench joblib dump and load functions, memmap modes."""
+    time_write = list()
```

Comment:
`time_write = []` I personally prefer to use litterals for list and dictionaries.